test/WebJobs.Script.Tests.E2E/TrackedAssert.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using Microsoft.ApplicationInsights;
+using Xunit;
+
+namespace WebJobs.Script.EndToEndTests
+{
+    public class TrackedAssert
+    {
+        private readonly TelemetryClient _client;
+
+        public TrackedAssert(TelemetryClient client)
+        {
+            _client = client;
+        }
+
+        public void Equals(string expected, string actual, [CallerMemberName] string testName = null, string assertionName = "")
+        {
+            RunAssert(testName, assertionName, () => Assert.Equal(expected, actual));
+        }
+
+        public void True(bool condition, [CallerMemberName] string testName = null, string assertionName = "")
+        {
+            RunAssert(testName, assertionName, () => Assert.True(condition));
+        }
+
+        private void RunAssert(string testName, string assertionName, Action assertion)
+        {
+            try
+            {
+                assertion();
+            }
+            catch (Exception exc)
+            {
+                TraceTestResult(testName, assertionName, false, exc);
+                throw;
+            }
+
+            TraceTestResult(testName, assertionName, true);
+        }
+
+        private void TraceTestResult(string testName, string assertionName, bool success, Exception exc = null)
```

Comment:
Short of writing our own runner, I didn't find a good way to get pass/fail per test, so I went with this. The `TestTraceAttribute` (`BeforeAfterTestAttribute`) unfortunately doesn't provide that information.